BCStudentSoftwareDevTeam/celts
# Change phone number edits

Issue: Make sure the behavior of phone number inputs are consistent across the application and make it so that if a user has made changes to a phone number and the click off without saving the phone number returns to how it was originally. 

________________________________________
Did:
- Created two new functions in base.js: setupPhoneNumber and processPhoneSetup.
- Changed the notification when the phone number is saved from a msgFlash to a msgToast.

________________________________________
Expected behavior when editing phone number:
-  On click of _"Edit"_ or on _focus_ of input: _"Edit"_ will become _"Save"_ and the phone input will be focused.
- On click of _"Save"_ with a VALID phone number: _"Save"_ will become _"Edit"_, the phone input will be un-focused, the new phone number will be saved, and a toast will appear in the bottom right confirming the new phone number has been saved.
- On click of _"Save"_ with an INVALID phone number: _"Save"_ will stay _"Save"_, the background of the phone input will turn red, and the phone input will stay focused.
- On _focus out_ of phone input without clicking _"Save"_: _"Save"_ will become _"Edit"_ and the contents of the phone input will restore the phone number to its original state before the input was focused. 

________________________________________
To test on User Profile:
- Make sure that the edit button behaves as expected when you are focusing/un-focusing the phone input.
- Delete part of the phone number then click out of the phone input without click the save button and make sure the phone number is restored to its origin state.
- Delete part of the phone number  and attempt to save it. It should not be allowed to save and the phone input bg color should turn red while still being focused. 
- Change the phone number and save it. Make sure the new phone number is actually saved and the msgToast pops up confirming the save. 

________________________________________
To test on SLC Instructor:
- Edit an SLC proposal and repeat all testing actions that were done on the User Profile on the Instructors phone number. 
- Add a second SLC instructor to the instructor table and make sure the phone numbers still behave as expected. 
- Remove all instructors from the instructor table and then add one back and make sure the phone number still behaves as expected.
